src/Model.cc

```diff
@@ -792,11 +796,34 @@ const NestedInclude *Model::InterfaceModelNestedIncludeByIndex(
 }
 
 /////////////////////////////////////////////////
-sdf::ElementPtr Model::ToElement() const
+std::string Model::Uri() const
+{
+  return this->dataPtr->uri;
+}
+
+/////////////////////////////////////////////////
+void Model::SetUri(const std::string &_uri)
+{
+  this->dataPtr->uri = _uri;
+}
+
+/////////////////////////////////////////////////
+sdf::ElementPtr Model::ToElement(bool _useUri) const
 {
+  if (_useUri && !this->dataPtr->uri.empty())
+  {
+    sdf::ElementPtr worldElem(new sdf::Element);
+    sdf::initFile("world.sdf", worldElem);
+
+    sdf::ElementPtr includeElem = worldElem->AddElement("include");
+    includeElem->GetElement("uri")->Set(this->Uri());
+    includeElem->GetElement("name")->Set(this->Name());
+    includeElem->GetElement("pose")->Set(this->RawPose());
+    return includeElem;
```

Comment:
Other elements are missing: `static`, `placement_frame`.

Reply:
Added `static` and `placement_frame` in 7e13033.